src/extension.ts

```diff
@@ -0,0 +1,140 @@
+import { from } from 'rxjs'
+import { filter, switchMap } from 'rxjs/operators'
+import * as sourcegraph from 'sourcegraph'
+import { resolveSettings, Settings } from './settings'
+
+/**
+ * Params derived from the document's URI.
+ */
+interface Params {
+    repo: string | null
+    file: string | null
+}
+
+interface SentryProject {
+    projectId: string
+    lineMatches: RegExp[]
+}
+
+const DECORATION_TYPE = sourcegraph.app.createDecorationType()
+const SETTINGSCONFIG = resolveSettings(sourcegraph.configuration.get<Settings>().value)
+
+export function activate(context: sourcegraph.ExtensionContext): void {
+    sourcegraph.workspace.onDidOpenTextDocument.subscribe(textDocument => {
+        const params: Params = getParamsFromUriPath(textDocument.uri)
+        const sentryProjects = SETTINGSCONFIG['sentry.projects']
+
+        // Retrieve the Sentry project that this document reports to.
+        const sentryProject = matchSentryProject(params, sentryProjects)
```

Comment:
This is business logic and should be moved outside of `activate()` and into a separate, testable function

Reply:
Will do in next PR.